core/codegen/src/lib.rs

```diff
@@ -58,8 +56,8 @@
 //! ROCKET_CODEGEN_DEBUG=1 cargo build
 //! ```
 
-#[macro_use] extern crate quote;
-extern crate proc_macro;
```

Comment:
This particular line also bumps the minimum required cargo version.

Reply:
This was suggested by `clippy`, I can revert this part, but Rocket requires Rust nightly, and I would say that most of the time Rust is installed with the same cargo version.

Reply:
I'm okay bumping the minimum version, but it has to be accompanied by a change to the corresponding `build.rs` file(s). I can do that when I merge, too.

Reply:
I have reverted it for now. I think that this PR should be about the dependencies. Maybe it would make sense to upgrade the syntax to 2018 edition, but that would include removing all `extern crate` references, and importing macros where they are used.

Reply:
Everything is already upgraded to the 2018 edition. `proc_macro` was a special case that still needed `extern crate` until a change in `cargo` around mid-December.